src/url-replacements.js

```diff
@@ -239,6 +242,14 @@ class UrlReplacements {
       return this.getTimingData_('navigationStart',
           'domContentLoadedEventStart');
     });
+
+    // Returns an identifier for the viewer.
+    this.set_('VIEWER', () => {
+      const viewerUrl = viewerFor(this.win_).getParam(VIEWPORT_ORIGIN_PARAM_) ||
```

Comment:
There's no real point of doing handshake in a native app - it's very easily spoofed and there's no way to confirm.

I'd expect `messagingOrigin_` to be always set in a iframed scenario. Native apps - we don't yet have a way to always provide it. Feel free to file a bug, e.g. "Propagate viewer info from native app".

Let's start with exposing messagingOrigin_ as a promise. Let's call it `getViewerOrigin()`. I'll provide other details in the PR review itself.